src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,91 @@
+package duke.storage;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Implements file storage.
+ *
+ * @author Audrey Felicio Anwar
+ */
+public class Storage {
+    private final File data;
+
+    /**
+     * Intializes a Storage object.
+     *
+     * @param filePath The location of the storage.
+     * @throws DukeException If directory not found.
+     */
+    public Storage(String filePath) throws DukeException {
+        try {
+            File data = new File(filePath);
+            if (!data.exists()) {
+                data.createNewFile();
+                assert data.exists() : " Directory cannot be made";
+            }
+            this.data = data;
+        } catch (IOException error) {
+            throw new DukeException(" I cannot find the directory :(");
+        }
+    }
+
+    /**
+     * Generates list of tasks from save file.
+     *
+     * @return List of tasks generated from save file.
+     * @throws DukeException If there is file reading error.
+     */
+    public ArrayList<Task> loadTasks() throws DukeException {
+        try {
+            ArrayList<Task> tasks = new ArrayList<>();
+            Scanner sc = new Scanner(data);

```

Comment:
I understand the "sc" represents the scanner variable, but maybe it is better to use something just like "scanner"?